### PR TITLE
Fix qualified category on rank status

### PR DESCRIPTION
**Pull request by Joehuu**
Qualified is actually number 3, and mod requests isn't a category at the moment.

### CODE REVIEW

**Comment by peppy** [comment]
Many of these are wrong (they are basd off the legacy stable api, which returns different status codes from the database).

Here's the correct values if you'd like to fix the others too:
```php
    const STATES = [
        'graveyard' => -2,
        'wip' => -1,
        'pending' => 0,
        'ranked' => 1,
        'approved' => 2,
        'qualified' => 3,
        'loved' => 4,
    ];
```

**Comment by Joehuu** [comment]
I'm confused, it works fine as is. The implementation was right, but `Qualified` was named `Mod Requests` for some reason.

I think those values you gave are for states(status of a map) not categories (i.e. ranked is a state, Ranked & Approved is a category; Any is a category that has all the states; WIP is not a category at all, it is part of the pending category.
TL;DR: I think the values you gave are states not categories b/c there is no `Favourites`, `Any`, or `My Maps` values. There is `WIP` which is not a category.

**Comment by peppy** [comment]
`RankStatus` was never intended to be categories. Please re-read my comment, where I tell you what the existing values where.

Please update it to match and add a new enum with the following:
```
        $statuses = [
            ['id' => 7, 'name' => trans('beatmaps.status.any')],
            ['id' => 0, 'name' => trans('beatmaps.status.ranked-approved')],
            ['id' => 1, 'name' => trans('beatmaps.status.approved')],
            ['id' => 3, 'name' => trans('beatmaps.status.qualified')],
            ['id' => 8, 'name' => trans('beatmaps.status.loved')],
            ['id' => 2, 'name' => trans('beatmaps.status.faves')],
            ['id' => 4, 'name' => trans('beatmaps.status.pending')],
            ['id' => 5, 'name' => trans('beatmaps.status.graveyard')],
            ['id' => 6, 'name' => trans('beatmaps.status.my-maps')],
        ];
```

Call it `BeatmapSearchCategory`

**Comment by Joehuu** [comment]
I think the file was made already.
https://github.com/ppy/osu/blob/master/osu.Game/Beatmaps/BeatmapSetOnlineStatus.cs

**Comment by peppy** [comment]
Okay, but this one still needs to be renamed

**Review by smoogipoo — DISMISSED** [review]
Looks good to me 👍

**Review by smoogipoo — APPROVED** [review]
